src/mscorlib/corefx/System/Globalization/CompareInfo.Unix.cs

```diff
@@ -26,7 +26,14 @@ internal CompareInfo(CultureInfo culture)
         private void InitSort(CultureInfo culture)
         {
             _sortName = culture.SortName;
-            _sortHandle = Interop.GlobalizationInterop.GetSortHandle(GetNullTerminatedUtf8String(_sortName));
+            Interop.GlobalizationInterop.ResultCode resultCode = Interop.GlobalizationInterop.GetSortHandle(GetNullTerminatedUtf8String(_sortName), out _sortHandle); 
+            if (resultCode != Interop.GlobalizationInterop.ResultCode.Success)
+            {
```

Comment:
> in case we return non success code, we already freed/disposed the sort handle

My comment wasn't because of a concern about leaking an underlying handle; it was because we've allocated a finalizable SafeHandle that will be left for finalization.